src/query/execute_statement.c

```diff
@@ -17724,3 +17758,361 @@ do_insert_checks (PARSER_CONTEXT * parser, PT_NODE * statement, PT_NODE ** class
 exit:
   return error;
 }
+
+MOP
+do_get_server_obj_id (DB_IDENTIFIER * server_obj_id, DB_OBJECT * server_class_mop, const char *server_name)
+{
+  return do_get_obj_id (server_obj_id, server_class_mop, server_name, "link_name");
+}
+
+int
+do_drop_server (PARSER_CONTEXT * parser, PT_NODE * statement)
+{
+  DB_OBJECT *server_class = NULL, *server_object = NULL;
+  DB_IDENTIFIER server_obj_id;
+  DB_VALUE class_name_val;
+  char *name;
+  int error = NO_ERROR;
+  int save;
+  bool au_disable_flag = false;
+
+  CHECK_MODIFICATION_ERROR ();
+
+  db_make_null (&class_name_val);
+  OID_SET_NULL (&server_obj_id);
+
+  server_class = sm_find_class (CT_DB_SERVER_NAME);
+  if (server_class == NULL)
+    {
+      error = ER_DBLINK_CATALOG_DB_SERVER_NOT_FOUND;
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, error, 0);
+      goto end;
+    }
+
+  name = (char *) statement->info.drop_server.server_name->info.name.original;
+  server_object = do_get_server_obj_id (&server_obj_id, server_class, name);
+  if (server_object == NULL)
+    {
+      if (statement->info.drop_server.if_exists)
+	{
+	  return NO_ERROR;
+	}
+      error = ER_DBLINK_SERVER_NOT_FOUND;
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, error, 1, name);
+      PT_ERRORmf (parser, statement, MSGCAT_SET_PARSER_RUNTIME, MSGCAT_RUNTIME_RT_SERVER_NOT_DEFINED, name);
+      goto end;
+    }
+
+  /* check if user is creator or DBA  */
+  error = au_check_serial_authorization (server_object);
```

Comment:
looks copy and paste fault

Reply:
Regardless of the function name, what is being processed is correct.
However, there is a risk of confusion, so I will separate the function name and rewrite it.